e2e/e2e_tests.py

```diff
@@ -0,0 +1,151 @@
+import os
+from time import sleep
+import unittest
+import subprocess
+
+try:
+    import requests
+    from bs4 import BeautifulSoup
+except ImportError:
+    requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+    subprocess.check_call("pip install -r {}".format(requirements_file).split(' '))
+    import requests
+    from bs4 import BeautifulSoup  # noqa ignore imported but unused
+
+from helpers.html_helpers import make_soup
+from helpers.zip_file_helpers import check_if_result_contains_data
+from helpers import docker_compose
+
+ADMIN_USER_FOR_TESTS = 'admin'
+ADMIN_PASSWORD_FOR_TESTS = 'admin'
+
+
+def _start_containers():
+    """
+    first clean up, then check for newer images and then start the containers.
+    Then wait 10 seconds to be certain the webapp is up and ready to receive.
+    """
+    docker_compose.clean()
+    docker_compose.pull()
+    docker_compose.start()
+    docker_compose.create_superuser_for_test(username=ADMIN_USER_FOR_TESTS, password=ADMIN_PASSWORD_FOR_TESTS)
+    sleep(10)
+
+
+def _stop_containers():
+    """
+    stop the containers and then cleanup
+    """
+    # docker_compose.stop()
+    # docker_compose.clean()
+
+
+class TestE2E(unittest.TestCase):
+    COOKIES = dict()
+    CSRF_TOKEN = ''
+
+    def _login(self, next=None):
+        login_url = 'http://localhost:8000/login/'
+        client = requests.session()
+        client.get(login_url)
+
+        csrf_token = client.cookies['csrftoken']
+        login_data = {
+            'password': ADMIN_PASSWORD_FOR_TESTS,
+            'username': ADMIN_USER_FOR_TESTS,
+            'csrfmiddlewaretoken': csrf_token,
+            'next': next,
+        }
+        r = client.post(login_url, data=login_data)
+        return {'request': r, 'client': client}
+
+    def _logout(self):
+        return requests.get('http://localhost:8000/logout/')
+
+    def _make_link(self, link):
+        host = 'http://localhost:8000{}'
+        return host.format(link)
+
+    def _download_and_test_zip_contents(self, client, download_link):
+        download_link = self._make_link(download_link)
+        r = client.get(download_link)
+        check_if_result_contains_data(r.content, self.assertGreater)
+
+    def test_server_is_running(self):
+        r = requests.get('http://localhost:8000/')
+        self.assertEqual(r.status_code, 200)
+
+    def test_login_denied_without_csrf_token(self):
+        login_data = {'username': ADMIN_USER_FOR_TESTS, 'password': ADMIN_PASSWORD_FOR_TESTS}
+        r = requests.post('http://localhost:8000/login/', data=login_data)
+        self.assertEqual(r.status_code, 403)
+
+    def test_login_success_with_csrf_token(self):
+        login = self._login(next='/orders/')
+        r = login['request']
+        soup = make_soup(r.content)
+        self.assertIsNotNone(soup.find(name='a', attrs={'href': '/logout/?next=/'}))
+        self.assertEqual(r.status_code, 200)
+        self._logout()
```

Comment:
It doesn't matter, if logout isn't called if something fails, since the cookie with the csrftoken will be invalid anyway. We can actually remove that code completely :)